test/e2e/e2e_test.go

```diff
@@ -296,8 +387,193 @@ var _ = Describe("Manager", Ordered, func() {
+			Eventually(
+				func() error {
+					key := plr.GetNamespacedName()
+					err := k8sClient.Get(ctx, key, plr)
+					if err != nil {
+						return err
+					}
+					if plr.Spec.Status != tekv1.PipelineRunSpecStatusPending {
+						return fmt.Errorf("PipelineRun status is %s and not Pending", plr.Spec.Status)
+					}
+
+					return nil
+				},
+				30*time.Second,
+				3*time.Second,
+			).Should(Succeed())
+		})
+
+		It("A matching workload was created for the PipelineRun", func(ctx context.Context) {
+			Eventually(func() error {
+				wl, err := GetOwnedWorkload(k8sClient, plr, ctx)
+				if err != nil {
+					return err
+				}
+				cond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadQuotaReserved)
+				if cond == nil {
+					return fmt.Errorf("Didn't find QuotaReserved condition for workload %s", wl.Name)
+				}
+
+				if cond.Status != metav1.ConditionFalse {
+					return fmt.Errorf("QuotaReserved Condition status isn't false (Quota Was reserved while it should have failed)")
+				}
+
+				if !strings.Contains(cond.Message, "insufficient quota for memory") {
+					return fmt.Errorf("Didn't find expected condition message")
+				}
+
+				return nil
+			},
+				15*time.Second,
+				3*time.Second,
+			).Should(Succeed())
+		})
+	})
 })
 
+func GetOwnedWorkload(k8sClient client.Client, plr *tekv1.PipelineRun, ctx context.Context) (*kueue.Workload, error) {
+	wlList := &kueue.WorkloadList{}
+	ownerKey := jobframework.GetOwnerKey(tekv1.SchemeGroupVersion.WithKind("PipelineRun"))
+	err := k8sClient.List(
+		ctx,
+		wlList,
+		client.MatchingFields{ownerKey: plr.Name},
+	)
+	if err != nil {
+		return nil, err
+	}
+	if len(wlList.Items) != 1 {
+		return nil, fmt.Errorf("found %d owners for the workload", len(wlList.Items))
+	}
+	wl := wlList.Items[0]
+	hasOwner, err := controllerutil.HasOwnerReference(wl.OwnerReferences, plr, k8sClient.Scheme())
+	if err != nil {
+		return nil, err
+	}
+	if !hasOwner {
+		return nil, fmt.Errorf("The workload owner doesn't match the pipelinerun %s", plr.Name)
+	}
+	return &wl, nil
+}
+
+func getK8sClientOrDie(ctx context.Context) client.Client {
+	scheme := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(tekv1.AddToScheme(scheme))
+	utilruntime.Must(kueue.AddToScheme(scheme))
+
+	cfg := ctrl.GetConfigOrDie()
+	k8sCache, err := cache.New(cfg, cache.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred(), "failed to create cache")
+
+	err = jobframework.SetupWorkloadOwnerIndex(
+		ctx,
+		k8sCache,
+		tekv1.SchemeGroupVersion.WithKind("PipelineRun"),
+	)
+
+	Expect(err).ToNot(HaveOccurred(), "failed to setup indexer")
+
+	k8sClient, err := client.New(
+		cfg,
+		client.Options{
+			Cache:  &client.CacheOptions{Reader: k8sCache},
+			Scheme: scheme,
+		},
+	)
+	Expect(err).ToNot(HaveOccurred(), "failed to create client")
+
+	go func() {
+		err := k8sCache.Start(ctx)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
Should we wait for cache to synch?

```suggestion
		if err := k8sCache.Start(ctx); err != nil {
			panic(err)
		}

		if synced := k8sCache.WaitForCacheSync(ctx); !synced {
			panic("failed waiting for cache to sync")
		}
```